lib/Pipes/FileContainer.cpp

```diff
@@ -97,8 +97,12 @@ void FileContainer::mergeBackImpl(FileContainer &&Container) {
 }
 
 llvm::Error FileContainer::storeToDisk(llvm::StringRef Path) const {
-  if (this->Path.empty())
+  if (this->Path.empty()) {
+    if (llvm::sys::fs::exists(Path))
+      llvm::sys::fs::remove(Path);
```

Comment:
What's the ratio for this? Seems surprising.

Reply:
the issue is that if the output binary has been produced and then it gets somehow invalidated, without this the file it refers too would not be erased. Thus at the second execution it would be reloaded. 

The issue in reading this may be that this->Path is not Path the argument of the function. this->Path is the tmp file, Path is the final destination of the file

Reply:
I noticed the difference between `this->Path` and `Path` (maybe rename the argument).

If *not* producing a file is an expected behavior of `storeToDisk`, I think it should be the same for all the `storeToDisk`, not just `FileContainer::storeToDisk`.
The idea should be 1) delete the file if exists, 2) output something if it needs the be produced.

Reply:
the difference is that other containers (llvm containers mostly) are not truly empty when their enumeration is empty. LLVMContainers have all the various helpers that stick around

Reply:
Mmmmmmh, I find this behavior surprising, but OK.

Reply:
Please add a comment about this.